src/lwc/geFormRenderer/geFormRenderer.js

```diff
@@ -711,7 +712,8 @@ export default class GeFormRenderer extends LightningElement{
 
                 dataImportFromFormState = this.saveableFormState();
             } catch (ex) {
-                this.handleAsyncWidgetError(ex);
+                const errors = [{ message: buildErrorMessage(ex) }];
+                this.handlePurchaseCallValidationErrors(errors);
```

Comment:
Not sure if we need to rename this function to be more generic or if I should be calling something else - this calls through to handleAsyncWidgetError() but does some additional transformation and checking and label replacement that is likely useful here. 

I'm also wondering if the call to `this.handleAsyncWidgetError(ex);` on line 673 of handleSave() should also do something similar - I think I've seen that come through as "unknown error" when there's an exception thrown, and using handlePurchaseXCallValidationErrors() or something similar would massage that in a way that gets it to display as we intend, I believe.

Reply:
Yeah, I think that makes sense. We should probably also just pull that entire try/catch block for tokenizing inside handleSave() into its own method entirely.